apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts

```diff
@@ -0,0 +1,328 @@
+import { html, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+import { nodeModulesAssetPath } from '../../lib/assets';
+
+export interface AdministratorQueryRunParams {
+  name: string;
+  sql: string;
+  params: string;
+  authn_user_id: string;
+  error: string | null;
+  result: AdministratorQueryResult | null;
+  formatted_date: string | null;
+}
+
+export interface AdministratorQueryResult {
+  rows: any[];
```

Comment:
I believe this could be a little more precisely typed:

```suggestion
  rows: Record<string, any>[];
```

This says "`rows` is an array of objects with string keys and values of any type".

Reply:
This has been refactored to a zod schema. Resolved in commit 2f39462.